pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -361,13 +352,17 @@ public Representation post(Representation entity) {
           // list of FileItems
           items = upload.parseRequest(getRequest());
 
-          for (final Iterator<FileItem> it = items.iterator(); it.hasNext() && !found;) {
-            final FileItem fi = it.next();
-            if (fi.getFieldName() != null) {
-              found = true;
-              dataFile = new File(tempDir, fi.getFieldName());
-              fi.write(dataFile);
+          for (FileItem fileItem : items) {
+            if (!found) {
+              if (fileItem.getFieldName() != null) {
+                found = true;
+                dataFile = new File(tempDir, fileItem.getFieldName());
+                fileItem.write(dataFile);
+              }
+            } else {
+              LOGGER.warn("Got extra file item while pushing segments: " + fileItem.getFieldName());
             }
+            fileItem.delete();
```

Comment:
From what i read on commons io and fileupload packages, and from what we have seen, the temporary file should be cleaned up. The documentation reads that the file is cleaned up when the object is garbage collected. I am not sure if deleting the file actively will cause something else to break. 

I saw some code in newer version of fileupload utils that deletes the file in their code.